### ISSUE TITLE
XDS pushes sent when endpoint values are not changing

### ISSUE THREAD

**Issue by ivantrips** [istio/istio#48025]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

Currently having a lot of XDS pushes which seem unnecessary as the underlying endpoint is not really changing 

Using the [postgres operator](https://github.com/zalando/postgres-operator) and the postgres pods are injected by istio to have mTLS from the app pods to postgres

By default patroni doesn't use a selector on the primary service, to avoid two pods having the same `master` selector and causing a load balance between two pods that are not the master, so they update the endpoint when is required.

Here is a definition of the underlaying endpoint, however neither the `hostname` or `ip` in the subsets are really changing, but the `metadata.annotations.renewTime` changes every 10 seconds

```yaml
apiVersion: v1
kind: Endpoints
metadata:
  annotations:
    acquireTime: "2023-11-21T23:15:04.300445+00:00"
    endpoints.kubernetes.io/last-change-trigger-time: "2023-11-19T14:45:07Z"
    leader: postgres-0
    optime: "346587227448"
    renewTime: "2023-11-23T20:33:59.191237+00:00"
    transitions: "36"
    ttl: "30"
  labels:
    application: spilo
    cluster-name: postgres
    spilo-role: master
  name: postgres
  namespace: postgres
subsets:
- addresses:
  - hostname: postgres-0
    ip: 10.40.35.83
    nodeName: worker-01
    targetRef:
      kind: Pod
      name: postgres-0
      namespace: postgres
  ports:
  - name: postgresql
    port: 5432
    protocol: TCP
```

Normally a lot of pushes are ok, but under high load we start to see SSL errors in the sidecar containers.

```
[2023-11-21T23:21:24.982Z] "- - -" 0 UF,URX - - "TLS_error:_268435588:SSL_routines:OPENSSL_internal:CLIENTHELLO_TLSEXT_268435646:SSL_routines:OPENSSL_internal:PARSE_TLSEXT" 0 0 9998 - "-" "-" "-" "-" "10.40.35.119:5432" outbound|5432||postgres.postgres.svc.cluster.local - 10.97.15.217:5432 10.40.19.154:34686 - -
[2023-11-21T23:21:24.982Z] "- - -" 0 UF,URX - - "TLS_error:_268435588:SSL_routines:OPENSSL_internal:CLIENTHELLO_TLSEXT_268435646:SSL_routines:OPENSSL_internal:PARSE_TLSEXT" 0 0 9998 - "-" "-" "-" "-" "10.40.35.119:5432" outbound|5432||postgres.postgres.svc.cluster.local - 10.97.15.217:5432 10.40.19.154:34660 - -
[2023-11-21T23:21:24.982Z] "- - -" 0 UF,URX - - "TLS_error:_268435588:SSL_routines:OPENSSL_internal:CLIENTHELLO_TLSEXT_268435646:SSL_routines:OPENSSL_internal:PARSE_TLSEXT" 0 0 9998 - "-" "-" "-" "-" "10.40.35.119:5432" outbound|5432||postgres.postgres.svc.cluster.local - 10.97.15.217:5432 10.40.19.154:34632 - -
[2023-11-21T23:21:24.983Z] "- - -" 0 UF,URX - - "TLS_error:_268435588:SSL_routines:OPENSSL_internal:CLIENTHELLO_TLSEXT_268435646:SSL_routines:OPENSSL_internal:PARSE_TLSEXT" 0 0 9998 - "-" "-" "-" "-" "10.40.35.119:5432" outbound|5432||postgres.postgres.svc.cluster.local - 10.97.15.217:5432 10.40.19.154:34724 - -
```

This gives us two questions
- Is there a way of not sending an XDS push to all the pods if the endpoint ip or hostname are not really changing?
- What are the limitations of the istiod XDS pushes in a cluster or multi-primary, multi-network environment 

### Version

```Text
$ istioctl version
1.15.7
$ kubectl version --short
Client Version: v1.27.2
Kustomize Version: v5.0.1
Server Version: v1.26.6-gke.1700
```

**Comment by ramaraochavali** [comment]
1.15 is unsupported now and in the recent versions we have moved away from Endpoints infavour of EndpointSlices

**Comment by ivantrips** [comment]
@ramaraochavali in which version was that moved?

i could take a look into the changelog and test a version that has that change

**Comment by ivantrips** [comment]
So after doing some testing, i found that the change you mentioned @ramaraochavali was done in 1.13
> Updated the control plane to read EndpointSlice instead of Endpoints for service discovery for Kubernetes 1.21 or later. To switch back to the old Endpoints based behavior set PILOT_USE_ENDPOINT_SLICE=false in istiod.

And also found that this is resolved just in a different way in Istio 1.19, apparently

> Updated DNS resolution when using ServiceEntries so that DNS for multi-network gateways will be resolved at the proxy instead of in the control plane.

Still need to test how this affects in general the SSL inconsistencies but looks better

**Comment by ivantrips** [comment]
Found that for some reason upgrading to 1.19 does reduce this massive amount of pushes, and improves a lot multi-network DNS discovery
